src/main/java/duke/TaskList.java

```
@@ -0,0 +1,129 @@
+ * deleting/adding tasks and marking tasks as done/undone
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
+
+    /**
+     * Constructor for a TaskList object
+     * @param data ArrayList of String that represents the data read in from the data text file by the Storage object.
+     */
+    public TaskList(ArrayList<String> data) {
+        this.taskList = new ArrayList<>();
+        for (String taskDataString : data) {
+            Task task = deconstructStringIntoTask(taskDataString);
+            this.taskList.add(task);
+        }
+    }
+
+    /**
+     * Constructor for a TaskList object where there is no data
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    private Task deconstructStringIntoTask(String line) {
+        String[] array = line.split("/", 2);
+        if (array[0].equals("T")) {
+            String[] arrayT = line.split("/", 3);
+            Todo task = new Todo(arrayT[2]);
+            if (arrayT[1].equals("1")) {
+                task.markAsDone();
+            }
+            return task;
+        } else if (array[0].equals("D")) {
+            String[] arrayD = line.split("/", 4);
+            LocalDateTime byDateTime = LocalDateTime.parse(arrayD[3]);
+            Deadline task = new Deadline(arrayD[2], byDateTime);
+            if (arrayD[1].equals("1")) {
+                task.markAsDone();
+            }
+            return task;
+        } else if (array[0].equals("E")) {
+            String[] arrayE = line.split("/", 5);
+            LocalDateTime fromDateTime = LocalDateTime.parse(arrayE[3]);
+            LocalDateTime toDateTime = LocalDateTime.parse(arrayE[4]);
+            Event task = new Event(arrayE[2], fromDateTime, toDateTime);
+            if (arrayE[1].equals("1")) {
+                task.markAsDone();
+            }
+            return task;
+        } else {
+            return null;
```

Instead of returning `null`, could throwing an exception help handle this situation more gracefully?